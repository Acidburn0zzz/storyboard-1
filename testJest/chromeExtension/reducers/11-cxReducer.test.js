/* eslint-env jest */
import reducer from '../../../lib/chromeExtension/reducers/cxReducer';

describe('cxReducer', () => {
  let state;
  beforeEach(() => {
    state = reducer(undefined, { type: '' });
  });

  it('01 should have correct initial state', () => {
    expect(state).toMatchSnapshot();
  });

  describe('02 DevTools <-> WsClient cx state', () => {
    it('01 should process CX_CONNECTED', () => {
      state = reducer(state, { type: 'CX_CONNECTED' });
      expect(state).toMatchSnapshot();
    });

    it('02 should process CX_DISCONNECTED', () => {
      state = reducer(state, { type: 'CX_CONNECTED' });
      state = reducer(state, { type: 'CX_DISCONNECTED' });
      expect(state).toMatchSnapshot();
    });
  });

  describe('03 WsClient <-> WsServer cx state', () => {
    it('01 should process WS_CONNECTED', () => {
      state = reducer(state, { type: 'WS_CONNECTED' });
      expect(state).toMatchSnapshot();
    });

    it('02 should process CX_DISCONNECTED', () => {
      state = reducer(state, { type: 'WS_CONNECTED' });
      state = reducer(state, { type: 'WS_DISCONNECTED' });
      expect(state).toMatchSnapshot();
    });
  });

  describe('04 login state', () => {
    it('01 should process LOGIN_REQUIRED (false)', () => {
      state = reducer(state, { type: 'LOGIN_REQUIRED', fLoginRequired: false });
      expect(state).toMatchSnapshot();
    });

    it('02 should process LOGIN_REQUIRED (true)', () => {
      state = reducer(state, { type: 'LOGIN_REQUIRED', fLoginRequired: true });
      expect(state).toMatchSnapshot();
    });

    it('03 should process LOGIN_STARTED', () => {
      state = reducer(state, { type: 'LOGIN_STARTED' });
      expect(state).toMatchSnapshot();
    });

    it('04 should process LOGIN_SUCCEEDED', () => {
      state = reducer(state, { type: 'LOGIN_SUCCEEDED', login: 'John' });
      expect(state).toMatchSnapshot();
    });

    it('05 should process LOGGED_OUT', () => {
      state = reducer(state, { type: 'LOGIN_SUCCEEDED', login: 'John' });
      state = reducer(state, { type: 'LOGGED_OUT' });
      expect(state).toMatchSnapshot();
    });
  });

  describe('05 filters', () => {
    it('01 should process SERVER_FILTER', () => {
      state = reducer(state, { type: 'SERVER_FILTER', filter: 'abcd:*' });
      expect(state).toMatchSnapshot();
    });

    it('02 should process LOCAL_CLIENT_FILTER', () => {
      state = reducer(state, { type: 'LOCAL_CLIENT_FILTER', filter: 'dcba:*' });
      expect(state).toMatchSnapshot();
    });
  });
});
