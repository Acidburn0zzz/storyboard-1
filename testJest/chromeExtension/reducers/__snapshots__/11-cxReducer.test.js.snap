exports[`cxReducer 01 should have correct initial state 1`] = `
Object {
  "cxState": "DISCONNECTED",
  "fLoginRequired": null,
  "localClientFilter": null,
  "login": null,
  "loginState": "LOGGED_OUT",
  "serverFilter": null,
  "wsState": "DISCONNECTED",
}
`;

exports[`cxReducer 02 DevTools <-> WsClient cx state 01 should process CX_CONNECTED 1`] = `
Object {
  "cxState": "CONNECTED",
  "fLoginRequired": null,
  "localClientFilter": null,
  "login": null,
  "loginState": "LOGGED_OUT",
  "serverFilter": null,
  "wsState": "DISCONNECTED",
}
`;

exports[`cxReducer 02 DevTools <-> WsClient cx state 02 should process CX_DISCONNECTED 1`] = `
Object {
  "cxState": "DISCONNECTED",
  "fLoginRequired": null,
  "localClientFilter": null,
  "login": null,
  "loginState": "LOGGED_OUT",
  "serverFilter": null,
  "wsState": "DISCONNECTED",
}
`;

exports[`cxReducer 03 WsClient <-> WsServer cx state 01 should process WS_CONNECTED 1`] = `
Object {
  "cxState": "DISCONNECTED",
  "fLoginRequired": null,
  "localClientFilter": null,
  "login": null,
  "loginState": "LOGGED_OUT",
  "serverFilter": null,
  "wsState": "CONNECTED",
}
`;

exports[`cxReducer 03 WsClient <-> WsServer cx state 02 should process CX_DISCONNECTED 1`] = `
Object {
  "cxState": "DISCONNECTED",
  "fLoginRequired": null,
  "localClientFilter": null,
  "login": null,
  "loginState": "LOGGED_OUT",
  "serverFilter": null,
  "wsState": "DISCONNECTED",
}
`;

exports[`cxReducer 04 login state 01 should process LOGIN_REQUIRED (false) 1`] = `
Object {
  "cxState": "DISCONNECTED",
  "fLoginRequired": false,
  "localClientFilter": null,
  "login": null,
  "loginState": "LOGGED_OUT",
  "serverFilter": null,
  "wsState": "DISCONNECTED",
}
`;

exports[`cxReducer 04 login state 02 should process LOGIN_REQUIRED (true) 1`] = `
Object {
  "cxState": "DISCONNECTED",
  "fLoginRequired": true,
  "localClientFilter": null,
  "login": null,
  "loginState": "LOGGED_OUT",
  "serverFilter": null,
  "wsState": "DISCONNECTED",
}
`;

exports[`cxReducer 04 login state 03 should process LOGIN_STARTED 1`] = `
Object {
  "cxState": "DISCONNECTED",
  "fLoginRequired": null,
  "localClientFilter": null,
  "login": null,
  "loginState": "LOGGING_IN",
  "serverFilter": null,
  "wsState": "DISCONNECTED",
}
`;

exports[`cxReducer 04 login state 04 should process LOGIN_SUCCEEDED 1`] = `
Object {
  "cxState": "DISCONNECTED",
  "fLoginRequired": null,
  "localClientFilter": null,
  "login": "John",
  "loginState": "LOGGED_IN",
  "serverFilter": null,
  "wsState": "DISCONNECTED",
}
`;

exports[`cxReducer 04 login state 05 should process LOGGED_OUT 1`] = `
Object {
  "cxState": "DISCONNECTED",
  "fLoginRequired": null,
  "localClientFilter": null,
  "login": null,
  "loginState": "LOGGED_OUT",
  "serverFilter": null,
  "wsState": "DISCONNECTED",
}
`;

exports[`cxReducer 05 filters 01 should process SERVER_FILTER 1`] = `
Object {
  "cxState": "DISCONNECTED",
  "fLoginRequired": null,
  "localClientFilter": null,
  "login": null,
  "loginState": "LOGGED_OUT",
  "serverFilter": "abcd:*",
  "wsState": "DISCONNECTED",
}
`;

exports[`cxReducer 05 filters 02 should process LOCAL_CLIENT_FILTER 1`] = `
Object {
  "cxState": "DISCONNECTED",
  "fLoginRequired": null,
  "localClientFilter": "dcba:*",
  "login": null,
  "loginState": "LOGGED_OUT",
  "serverFilter": null,
  "wsState": "DISCONNECTED",
}
`;
